### ISSUE TITLE
qname is not handled case insensitive in received NOTIFYs

### ISSUE THREAD

**Issue by klaus3000** [PowerDNS/pdns#4575]
at least 3.4.10 has this bug

workaround:

```
--- pdns-3.4.10/pdns/packethandler.cc.orig      2016-10-13 12:10:05.064357997 +0000
+++ pdns-3.4.10/pdns/packethandler.cc   2016-10-13 18:43:04.508360268 +0000
@@ -766,7 +766,7 @@
   DNSBackend *db=0;
   DomainInfo di;
   di.serial = 0;
-  if(!B.getDomainInfo(p->qdomain, di) || !(db=di.backend)) {
+  if(!B.getDomainInfo(toLower(p->qdomain), di) || !(db=di.backend)) {
     L<<Logger::Error<<"Received NOTIFY for "<<p->qdomain<<" from "<<p->getRemote()<<" for which we are not authoritative"<<endl;
     return trySuperMaster(p);
   }
@@ -778,7 +778,7 @@
       return RCode::Refused;
     }
   }
-  else if(!db->isMaster(p->qdomain, p->getRemote())) {
+  else if(!db->isMaster(toLower(p->qdomain), p->getRemote())) {
     L<<Logger::Error<<"Received NOTIFY for "<<p->qdomain<<" from "<<p->getRemote()<<" which is not a master"<<endl;
     return RCode::Refused;
   }
```

**Comment by rgacogne** [comment]
This is already fixed in 4.0.0. This should be fixed in the backend, because they are expected to "get queries in aNy CAsE" according to the documentation. Proposed fix in #4590.

**Comment by pieterlexis** [comment]
Fixed in #4590
